libcloudforensics/gcp_test.py

```diff
@@ -16,28 +16,343 @@
+    mock_gce_api.return_value.disks.return_value.aggregatedList.return_value.execute.return_value = None
+    mock_gce_operation.return_value = {
+      # See https://cloud.google.com/compute/docs/reference/rest/v1/disks/aggregatedList
+      # for complete structure
+      'items': {
+        0: {
+          'disks': [
+            {
+              'name': FAKE_BOOT_DISK.name,
+              'zone': '/' + FAKE_BOOT_DISK.zone
+            }
+          ]
+        },
+        1: {
+          'disks': [
+            {
+              'name': FAKE_DISK.name,
+              'zone': '/' + FAKE_DISK.zone
+            }
+          ]
+        }
+      }
+    }
+    disks = FAKE_ANALYSIS_PROJECT.list_disks()
+    self.assertEqual(len(disks), 2)
+    self.assertTrue(FAKE_DISK.name in disks and FAKE_BOOT_DISK.name in disks)
+    self.assertEqual(disks[FAKE_DISK.name]['zone'], FAKE_DISK.zone)
+    self.assertEqual(disks[FAKE_BOOT_DISK.name]['zone'], FAKE_BOOT_DISK.zone)
+
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.list_instances')
+  def test_get_instance(self, mock_list_instances):
+    """Test that an instance of a project can be found. """
+    mock_list_instances.return_value = {
+      FAKE_INSTANCE.name: {
+        'zone': FAKE_INSTANCE.zone
+      }
+    }
+    found_instance = FAKE_ANALYSIS_PROJECT.get_instance(FAKE_INSTANCE.name, FAKE_INSTANCE.zone)
+    self.assertIsInstance(found_instance, gcp.GoogleComputeInstance)
+    self.assertEqual(found_instance.project, FAKE_INSTANCE.project)
+    self.assertEqual(found_instance.name, FAKE_INSTANCE.name)
+    self.assertEqual(found_instance.zone, FAKE_INSTANCE.zone)
+    self.assertEqual(found_instance._data, FAKE_INSTANCE._data)
+    self.assertRaises(RuntimeError, FAKE_ANALYSIS_PROJECT.get_instance, 'non-existent-instance')
+
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.list_disks')
+  def test_get_disk(self, mock_list_disk):
+    """Test that a disk of an instance can be found. """
+    mock_list_disk.return_value = {
+      FAKE_DISK.name: {
+        'zone': FAKE_DISK.zone
+      }
+    }
+    found_disk = FAKE_ANALYSIS_PROJECT.get_disk(FAKE_DISK.name)
+    self.assertIsInstance(found_disk, gcp.GoogleComputeDisk)
+    self.assertEqual(found_disk.project, FAKE_DISK.project)
+    self.assertEqual(found_disk.name, FAKE_DISK.name)
+    self.assertEqual(found_disk.zone, FAKE_DISK.zone)
+    self.assertRaises(RuntimeError, FAKE_ANALYSIS_PROJECT.get_disk, 'non-existent-disk')
+
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.gce_operation')
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.gce_api')
+  def test_create_disk_from_snapshot(self, mock_gce_api, mock_gce_operation):
+    """Test the creation of a disk from a snapshot. """
+    mock_gce_api.return_value.disks.return_value.insert.return_value.execute.return_value = None
+    mock_gce_operation.return_value = None
+
+    # create_disk_from_snapshot(snapshot)
```

Comment:
If you're gonna add a comment, can you be a bit more explicit? Does this mean that you're just testing this particular function?

Reply:
Addressed. The comment is just here to showcase how the method is called since some parameters are optional. I reflected this more explicitly.